ere-guests/libs/src/blobs.rs

```diff
@@ -0,0 +1,27 @@
+//! Utilities for working with blob versioned hashes in Ethereum blocks.
+//!
+//! This module is currently used as a temporary workaround to allow proper validation of CL kzg blob commitments
+//! against the execution block, until this is properly implemented as a formal EL header field.
+
+use alloy_genesis::ChainConfig;
+use k256::sha2::{Digest, Sha256};
+use reth_ethereum_primitives::Block;
+
+/// Calculates the hash of all versioned hashes in a block if Cancun is active at the block's number and timestamp.
+pub fn calculate_versioned_hashes_hash(
+    chain_config: &ChainConfig,
+    block: &Block,
+) -> Option<[u8; 32]> {
+    chain_config
+        .is_cancun_active_at_block_and_timestamp(block.number, block.timestamp)
+        .then_some(
+            Sha256::digest(block.body.blob_versioned_hashes_iter().fold(
+                Vec::new(),
+                |mut acc, hash| {
+                    acc.extend_from_slice(&hash.0);
+                    acc
+                },
+            ))
+            .into(),
+        )
```

Comment:
TL;DR:
- If `Cancun` isn't activated, the public input is `None`.
- If it is, then we do `sha256(tx1.blob_hashes || tx2.blob_hashes || ...)`.

The reason why I chose `sha256` instead of `keccak` is to be aligned with the same rationale used in [EIP-7685 for request hashes](https://eips.ethereum.org/EIPS/eip-7685#block-header)... which I think make this decision since `sha256` is friendly for CL compared to `keccak` which is aligned what we are trying to do there too. (Until we have an endgame solution, and prob remove this module)

Reply:
Noting here that since blob hashes are fixed length, we don't need a length prefix, ie we are not in the situation where we could be hashing ("ab", c") or ("a", "bc") in both cases the hash would be hash("abc") which is ambiguous.

Reply:
Could we check if most zkVMs have a sha256 precompile? 

(given this grows with number of blobs, it could become quite expensive)

Reply:
> Noting here that since blob hashes are fixed length, we don't need a length prefix, ie we are not in the situation where we could be hashing ("ab", c") or ("a", "bc") in both cases the hash would be hash("abc") which is ambiguous.

Ah yes, forgot to clarify this. This is why I didn't do the "double hashing" done in the EIP-7685 since in that case the `data` isn't fixed length so had that risk.

> Could we check if most zkVMs have a sha256 precompile?

Good call, I'll check and report back.

Reply:
Confirmed most support it, for reference it is easier to see [here](https://github.com/eth-act/zkevm-benchmark-workload/blob/master/precompile-patches)